itou/cities/management/commands/sync_cities.py

```diff
@@ -61,6 +91,77 @@ class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument("--wet-run", dest="wet_run", action="store_true")
 
+    def handle_deletions(self, cities_removed_by_api):
+        relations_to_fix = [
+            (JobDescription, "location"),
+            *((model, "insee_city") for model in AddressMixin.__subclasses__()),
+        ]
+        model_to_refill_infos = {}
+        for model, field_name in relations_to_fix:
+            refill_infos = {
+                pk: (previous_insee_code, last_synced_at)
+                for pk, previous_insee_code, last_synced_at in model.objects.filter(
+                    **{f"{field_name}__code_insee__in": cities_removed_by_api}
+                ).values_list("pk", f"{field_name}__code_insee", f"{field_name}__last_synced_at")
+            }
+            model.objects.filter(pk__in=refill_infos).update(**{field_name: None})
```

Comment:
Je pense que ça be fonctionne pas car Django va pas aimer lors du `.delete()` mais désactiver les contraintes des FK aurais je pense eu l'avantage d'obliger à ce que les anciennes villes soient migrées et donc de ne pas créer de "trou" :grin:.